src/listeners/messageCreate.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable operator-linebreak */
+/**
+ * "messageCreate" event listener for the bot.
+ */
+
+import { Client, Message } from 'discord.js';
+import {
+  extractNotionPageIdFromTreadByChannel,
+  getChatOpenAIPromptResponse,
+  getFormattedPrompt,
+  SUPPORT_IDENTIFIER,
+  updateNotionSupportTicketsDBEntry,
+} from '../utils';
+
+export default (client: Client): void => {
+  client.on('messageCreate', async (message: Message) => {
+    const msgChannel = await client?.channels
+      .fetch(message?.channelId)
+      .then((messageChannel: any) => {
+        console.log(messageChannel?.name);
+        return messageChannel;
+      })
+      .catch(console.error);
+
+    // Indicates that the community member is asking a software related question.
+    const containSupportIdentifier = message?.content?.includes('#question');
+    if (containSupportIdentifier) {
+      const notionPageId: string = await extractNotionPageIdFromTreadByChannel(msgChannel);
+      // We need to remove all text before the #question identifier.
+      const indexOfQuestionIdentifier = message?.content?.indexOf('#question');
+      const communityMemberMessage = message?.content?.slice(
+        indexOfQuestionIdentifier + SUPPORT_IDENTIFIER.length,
+      );
+
+      const formattedPrompt = getFormattedPrompt(communityMemberMessage);
+      const openAIResponse = await getChatOpenAIPromptResponse(formattedPrompt);
+      await updateNotionSupportTicketsDBEntry(
+        notionPageId,
+        [
+          {
+            message: `
+              Qtn Asked: ${formattedPrompt}
+              --------------------
+              Response: ${openAIResponse}
+              --------------------
+            `,
+            author: 'GFC Community Software Sparring Partner',
+          },
+        ],
+        false,
+      );
+      await message?.reply(
+        'Your question has been sent to the GFC Community Software Sparring Partner! Please check notion for the response.',
```

Comment:
whats a `GFC Community Software Sparring Partner`? and who is it? the bot? a community member? chatgpt system?

Reply:
@borolepratik please see this [notion page](https://www.notion.so/gitfitcode/GFC-AI-Professional-Software-Sparring-Partner-PSSP-53197f866c014d8595e0db5e1bf2f336)